### PR TITLE
fix(beads): align agent bead prefixes and force multi-hyphen IDs

### LINKED ISSUE #48: Agent bead creation fails with prefix mismatch after `hq-` town beads change
## Bug Description

Agent beads (witness, refinery, mayor, deacon) fail to create during `gt rig add` because they use hardcoded `gt-` prefix but are stored in town beads which now use `hq-` prefix. This breaks the basic workflow of adding rigs to a fresh Gas Town installation.

## Steps to Reproduce

1. Install Gas Town from latest source (commit `386dbf8` or later with `hq-` prefix change)

``` sh
git clone https://github.com/steveyegge/gastown.git
cd gastown
go install ./cmd/gt
```
2. Initialize a fresh Gas Town HQ:

``` sh
gt install ~/gt
cd ~/gt
git init  # needed for repo fingerprint
```

3. Add any rig:

``` sh
gt rig add gastown git@github.com:jpetitcolas/gastown.git --prefix gastown
```

## Expected Behavior

The rig should be created successfully with all agent beads (witness, refinery) created in town beads database without errors.

## Actual Behavior

Rig creation succeeds but agent bead creation fails with prefix mismatch error:

> Warning: Could not create agent beads: creating gt-gastown-witness: bd create --json --id=gt-gastown-witness --type=agent --title=Witness for gastown - monitors polecat health and
  progress. --description=Witness for gastown - monitors polecat health and progress.

  role_type: witness
  rig: gastown
  agent_state: idle
  hook_bead: null
  role_bead: gt-witness-role
  cleanup_status: null
  active_mr: null
  notification_level: null: Error: prefix mismatch: database uses 'hq' but you specified 'gt' (use --force to override)

  The town beads database (~/gt/.beads/) was initialized with hq- prefix (correct), but the code tries to create agent beads with gt- prefix (incorrect

## Environment

  - OS: Linux (WSL2) - also reproducible on other platforms
  - Go version: go1.25.5
  - Gas Town version: Built from commit 386dbf8 (latest main)
  - bd version: 0.43.0
  - tmux version: Not applicable to this issue

## Logs / Error Output

```
jpetitcolas@jpetitcolas-per:~/gt$ gt rig add gastown git@github.com:jpetitcolas/gastown.git --prefix gastown
Creating rig gastown...
  Repository: git@github.com:jpetitcolas/gastown.git
  Detected existing beads prefix 'gt' from source repo
  Warning: Could not create agent beads: creating gt-gastown-witness: bd create --json --id=gt-gastown-witness --type=agent --title=Witness for gastown - monitors polecat health and progress. --description=Witness for gastown - monitors polecat health and progress.

role_type: witness
rig: gastown
agent_state: idle
hook_bead: null
role_bead: gt-witness-role
cleanup_status: null
active_mr: null
notification_level: null: Error: prefix mismatch: database uses 'hq' but you specified 'gt' (use --force to override)

✓ Rig created in 10.0s
```

## Additional Context

Commit [6081847b](https://github.com/steveyegge/gastown/commit/6081847bde05b217d75e3b67033463bb8e0cd905) changed town beads from `gm-` to `hq-` prefix, but the agent bead ID functions were not updated:

``` go
  // internal/beads/beads.go - These still hardcode "gt" prefix:
  func MayorBeadID() string {
      return "gt-mayor"  // Should be "hq-mayor"
  }

  func DeaconBeadID() string {
      return "gt-deacon"  // Should be "hq-deacon"
  }

  func WitnessBeadID(rig string) string {
      return WitnessBeadIDWithPrefix("gt", rig)  // Should use "hq"
  }

  func RefineryBeadID(rig string) string {
      return RefineryBeadIDWithPrefix("gt", rig)  // Should use "hq"
  }
```

### LINKED ISSUE #218: Agent bead IDs use hardcoded 'gt-' prefix instead of rig's configured prefix
## Bug Description

When creating a new rig with `gt rig add` or adding crew with `gt crew add`, the commands attempt to create agent beads (witness, crew) with a hardcoded `gt-` prefix instead of using the rig's configured beads prefix.

## Steps to Reproduce

```bash
gt rig add project_ideas https://git.example.com/user/project-ideas.git
# Rig is created with prefix: pi

gt crew add surveyor --rig project_ideas
```

## Expected Behavior

Agent beads should use the rig's configured prefix:
- Witness: `pi-project_ideas-witness`
- Crew: `pi-project_ideas-crew-surveyor`

## Actual Behavior

Agent beads are created with hardcoded `gt-` prefix:
- Witness: `gt-project_ideas-witness`
- Crew: `pi-project_ideas-crew-surveyor` (crew uses correct prefix but still fails)

This causes errors because `gt-` isn't in routes.jsonl:

```
Error: prefix mismatch: database uses 'hq' but you specified 'gt' (use --force to override)
```

and

```
Error: no beads database found
```

## Environment

- gastown version: 0.2.1
- Routes configured: `bi-` → bigmuff, `pi-` → project_ideas, `hq-` → town

## Workaround

Manually create agent beads with the correct rig prefix:

```bash
bd create --id=pi-project_ideas-witness --type=agent --title="Witness for project_ideas" --description="..."
bd create --id=pi-project_ideas-crew-surveyor --type=agent --title="Crew worker surveyor" --description="..."
```

## Suggested Fix

The `gt rig add` and `gt crew add` commands should read the rig's configured prefix from the beads database/config and use that when constructing agent bead IDs, rather than using a hardcoded `gt-` prefix.

### PR DESCRIPTION
**Pull request by johnjhughes**
## Summary
Fix multi‑hyphen bead ID creation and rig‑prefix lookups, and make a macOS temp‑path test reliable cross‑platform.

## Related Issue
Fixes #48
Fixes #218

## Changes
- add `NeedsForceForID` and pass `--force` for multi‑hyphen deterministic IDs (agent/rig/convoy/formula)
- use rig prefix for polecat bead lookup; fall back to rigs.json when routes.jsonl is missing
- add tests for `NeedsForceForID` and rigs.json prefix fallback
- normalize temp path comparisons in `costs_workdir_test.go` (`/var` vs `/private/var` on macOS)

## Testing
- [x] Unit tests pass (`PATH=/tmp:$PATH BEADS_NO_DAEMON=1 go test ./...`)
- [x] Build passes (`go build ./...`)
- [x] Tests also run successfully on Ubuntu and macOS
- [ ] Manual testing performed

## Checklist
- [x] Code follows project style (gofmt applied)
- [ ] Documentation updated (not needed)
- [x] No breaking changes (or documented in summary)

## Dependency
This PR is based on #483 (build/test fixes). Please merge #483 first or allow these commits to be squashed together; **Update (2026‑01‑15):** rebased onto `upstream/main` at `20effb0a`, so it should now merge cleanly.

### CODE REVIEW

File: internal/beads/force.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import "strings"
+
+// NeedsForceForID returns true when a bead ID uses multiple hyphens.
+// Newer bd versions infer the prefix from the last hyphen, which breaks
+// system bead IDs like "st-stockdrop-polecat-nux" and "hq-cv-abc".
+// Those IDs are valid in Gas Town, so we pass --force when creating them.
+func NeedsForceForID(id string) bool {
+	return strings.Count(id, "-") > 1
+}
```
[the remark this answers is not in the document]
Comment (johnjhughes) [reply internal/beads/force.go:11]: Added test in response to comment.

**Comment by johnjhughes** [comment]
Thanks for the review — now that #483 has merged, I rebased onto `upstream/main` (now `20effb0a`, previously `42d9890e`) so this PR is scoped only to the bead fix; after that rebase, I added a small second commit to address minor upstream regressions (see update below).

## What changed since your review

- **Rebased on latest main** to pick up #483 (squash) and current upstream head.
- **Added a second, minimal fix commit** to address minor issues introduced upstream that broke the build or caused flaky tests after the rebase: `fix(mail_queue,checkpoint): remove invalid nil check and fix IsStale boundary` — **Update (2026‑01‑15):** upstream fixed/refactored the mail_queue path, so that part was dropped on rebase; this commit now only contains the checkpoint boundary fix.
  - `internal/cmd/mail_queue.go`: removed `queueCfg == nil` because `QueueConfig` is a struct; the nil check is invalid and was breaking builds. **Update:** this mail_queue change no longer applies after rebase.
- `internal/checkpoint/checkpoint.go`: changed `IsStale` from `>` to `>=` so a checkpoint is stale when age equals or exceeds the threshold, matching the `TestIsStale/exactly_threshold` expectation and reducing flakiness at the boundary.

## Reasoning

- **Checkpoint boundary change:** matches the test’s intended semantics and removes a flaky edge case at the equality boundary.
- **Mail queue fix:** `!ok` already covers missing config entries; the nil check is both invalid and unnecessary. Removing it restores correct behavior and builds cleanly. **Update:** mail_queue change removed after rebase due to upstream refactor.

## Tests / build

- **Ubuntu (Linux) + macOS:** `go test ./...` and `go build ./...` both pass (ran locally on those platforms); **Update (2026‑01‑15):** re-ran after rebase and they still pass.
- **CI:** not run yet because I don’t have access to approve workflows; a maintainer needs to approve CI.

**Comment by djdefi** [comment]
🙌 Would love to see this merged! Running into the same issue with a `djdefi-ops` prefix. The hyphenated prefix causes `gt sling` to fail with:
```
Error: prefix mismatch: database uses 'djdefi-ops' but you specified 'djdefi'
```
This PR looks like exactly the fix needed.

**Comment by djdefi** [comment]
✅

Built from your branch and tested with hyphenated prefix `djdefi-ops`:
```
gt sling djdefi-ops-lmp --agent copilot
✓ Work attached to hook (status=hooked)
```

No more `prefix mismatch: database uses 'djdefi-ops' but you specified 'djdefi'` error! 🎉

Minor: convoy creation still shows an `hq-cv-` prefix warning (likely legacy records), but the core sling functionality works.

**Comment by johnjhughes** [comment]
Rebased again onto `upstream/main` (20effb0a) on 2026‑01‑15.

Upstream has since fixed the minor mail_queue issue that was in my second commit, so that commit is now limited to the checkpoint boundary fix (`IsStale: >` → `>=`) to avoid conflicts and ensure the core multi‑hyphen bead fix can land cleanly.

Tests pass after rebase:
- `go test ./...`
- `go build ./...`

**Review by steveyegge — APPROVED** [review]
Code looks good - comprehensive fix for multi-hyphen bead ID handling.

**However, no CI checks have run on this branch.** Please push an empty commit to trigger CI:

```bash
git commit --allow-empty -m "trigger CI"
git push
```

Once CI passes, this can be merged.
